File: 0119. Pascal's Triangle II/0119.java
// O(n) space complexity
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> row = new ArrayList<>();
        row.add(1); // first element is always 1
        
        for(int i=0; i<rowIndex; i++){
            for(int j=i; j>0; j--){
                row.set(j, row.get(j) + row.get(j-1));
            }
            row.add(1);
        }

        return row;
    }
}

File: 0119. Pascal's Triangle II/0119.cpp
// memory allocated for the vector is rowIndex+1
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row(rowIndex+1);
        row[0]=1;
        for(int i=0; i<rowIndex; i++){
            for(int j=i;j>0;j--) row[j]=row[j]+row[j-1];
            row[i+1]=1;
        }
        return row;
    }
};